src/tests/ftest/dfuse/bash.py

```diff
@@ -64,109 +57,108 @@ def run_bashcmd(self, il_lib=None, compatible_mode=False):
+            f"if (( filesize != {dd_count}*{dd_blocksize} )); then exit 1; fi",
+            f"cp -r {abs_file_path1} {abs_file_path2}",
+            f"diff {abs_file_path1} {abs_file_path2}",
+            f"cmp --silent {abs_file_path1} {abs_file_path2}",
+            f"rm {abs_file_path2}",
+            f"mv {abs_file_path1} {abs_file_path2}",
+            f"ls -al {abs_file_path2}",
+            f"ls -al {abs_dir_path}/.",
+            f"ls -al {abs_dir_path}/..",
+            f"rm {abs_file_path2}",
+            f"rmdir {abs_dir_path}",
+            f"wc -l {fuse_root_dir}/src.c",
+            f"xxd {fuse_root_dir}/src.c",
+            f"ln -s src.c {link_name}",
+            f"readlink {link_name}",
+            f"realpath {link_name}",
+            f"head {fuse_root_dir}/src.c",
+            f"tail {fuse_root_dir}/src.c",
+            # f'more {fuse_root_dir}/src.c', # more hangs over ssh somehow
+            f"dos2unix {fuse_root_dir}/src.c",
+            f"gcc -o {fuse_root_dir}/output {fuse_root_dir}/src.c",
+            f"size {fuse_root_dir}/output",
+            f"readelf -s {fuse_root_dir}/output",
+            f"strip -s {fuse_root_dir}/output",
+            f"g++ -o {fuse_root_dir}/output {fuse_root_dir}/src.c",
+            f"gcc -c -o {fuse_root_dir}/obj_a.o {fuse_root_dir}/src_a.c",
+            f"gcc -c -o {fuse_root_dir}/obj_b.o {fuse_root_dir}/src_b.c",
+            f"ar -rc {fuse_root_dir}/lib.a {fuse_root_dir}/obj_a.o " f"{fuse_root_dir}/obj_b.o",
+            f"objdump -d {fuse_root_dir}/obj_a.o",
+            f"grep print {fuse_root_dir}/src.c",
+            "awk '{print $1}' " + fuse_root_dir + "/src.c",
+            f"base64 {fuse_root_dir}/src.c",
+            f"md5sum {fuse_root_dir}/src.c",
+            f"cksum {fuse_root_dir}/src.c",
+            f"bzip2 -z {fuse_root_dir}/lib.a",
+            f"chmod u-r {fuse_root_dir}/lib.a.bz2",
+            'fio --readwrite=randwrite --name=test --size="2M" --directory '
+            f'{fuse_root_dir}/ --bs=1M --numjobs="4" --ioengine=psync '
+            "--group_reporting --exitall_on_error --continue_on_error=none",
+            f'curl "https://www.google.com" -o {fuse_root_dir}/download.html',
+        ]
+        for cmd in commands:
+            try:
+                # execute bash cmds
+                ret_code = general_utils.pcmd(self.hostlist_clients, env_str + cmd, timeout=120)
+                if 0 not in ret_code:
+                    error_hosts = NodeSet(
+                        ",".join(
+                            [
+                                str(node_set)
+                                for code, node_set in list(ret_code.items())
+                                if code != 0
+                            ]
+                        )
+                    )
+                    raise CommandFailure(
+                        f"Error running '{cmd}' on the following hosts: {error_hosts}"
+                    )
```

Comment:
Pre-existing, but FYI we have newer, better function called `run_remote` that works like
```
result = run_remote(self.log, self.hostlist_clients, cmd)
if not result.passed:
    self.fail(f"... failed on {result.failed_hosts}")
```
Which eliminates this nasty logic to get a nodeset